CompilerSource/ipc/Path.fbs

```diff
@@ -0,0 +1,23 @@
+
+struct PathPoint {
+  x:int;
+  y:int;
+  speed:int;
+}
+
+table Path {
+  name:string;
+  id:int;
+
+  smooth:bool;
+  closed:bool;
+  precision:int;
+
+  //background_room_id:int; //This was used by LGM rather than ENIGMA; it doesn't need to be here.
+  snap_x:int; //TODO: Find out if these are needed and decide on keeping them
+  snap_y:int;
+
+  points:[PathPoint];

```

Comment:
This here is going to be a vector basically, is that ok? I can't find anywhere that FlatBuffers has a plain-old array field type. Because of this, we don't need to write the pointCount, should we anyway? Also consider that Image will have a vector of bytes. Is there any performance overhead involved here from not using plain arrays?

Reply:
A vector is great, and we should definitely *not* write the point count. The awesome thing about protobuf/flatbuf is that we don't have to worry about Java vector vs C++ vector vs JavaScript array—it just works. So let it just work.

I suspect the performance overhead can be eliminated using some kind of Reserve method, or at worst, AddAll. I don't know enough about FlatBuf to say for sure, though—proto doesn't have this problem (or rather, it always has this problem in the same way), because it always copies.